x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/effected_policy_select/effected_policy_select.tsx

```diff
@@ -0,0 +1,151 @@
+      (suffix): string | undefined => {
+        if (dataTestSubj) {
+          return `${dataTestSubj}-${suffix}`;
+        }
+      },
+      [dataTestSubj]
+    );
+
+    const selectableOptions: EffectedPolicyOption[] = useMemo(() => {
+      const isPolicySelected = selected.reduce<{ [K: string]: PolicyData }>(
+        (idToPolicyData, policy) => {
+          idToPolicyData[policy.id] = policy;
+          return idToPolicyData;
+        },
+        {}
+      );
+
+      return options
+        .map<EffectedPolicyOption>((policy) => ({
+          label: policy.name,
+          policy,
+          checked: isPolicySelected[policy.id] ? 'on' : undefined,
+          disabled: isGlobal,
+        }))
+        .sort(({ label: labelA }, { label: labelB }) => labelA.localeCompare(labelB));
+    }, [isGlobal, options, selected]);
+
+    const handleOnPolicySelectChange = useCallback<
+      Required<EuiSelectableProps<OptionPolicyData>>['onChange']
+    >((currentOptions) => {
+      setSelectionState((prevState) => ({
+        ...prevState,
+        selected: currentOptions.filter((opt) => opt.checked).map((opt) => opt.policy),
+      }));
+    }, [])!;
+
+    const handleGlobalSwitchChange: EuiSwitchProps['onChange'] = useCallback(
+      ({ target: { checked } }) => {
+        setSelectionState((prevState) => ({ ...prevState, isGlobal: checked }));
+      },
+      []
+    );
+
+    const listBuilderCallback: EuiSelectableProps['children'] = useCallback((list) => list, []);
+
+    // Anytime selection state is updated, call `onChange`
+    useEffect(() => {
+      onChange(selectionState);
+    }, [onChange, selectionState]);
+
+    return (
+      <>
+        <EuiFormRow
+          fullWidth
+          label={i18n.translate(
+            'xpack.securitySolution.trustedapps.policySelect.globalSectionTitle',
+            {
+              defaultMessage: 'Global application',
+            }
+          )}
+        >
+          <EuiSwitch
+            label={i18n.translate(
+              'xpack.securitySolution.trustedapps.policySelect.globalSwitchTitle',
+              {
+                defaultMessage: 'Apply trusted application globally',
+              }
+            )}
+            checked={selectionState.isGlobal}
+            onChange={handleGlobalSwitchChange}
+            data-test-subj={getTestId('globalSwitch')}
+          />
+        </EuiFormRow>
+        <EuiFormRow
+          fullWidth
+          label={i18n.translate('xpack.securitySolution.policySelect.policySpecificSectionTitle', {
+            defaultMessage: 'Apply to specific endpoint policies',
+          })}
+        >
+          <EuiSelectable<OptionPolicyData>
+            {...otherSelectableProps}
+            options={selectableOptions}
+            listProps={listProps || DEFAULT_LIST_PROPS}
+            onChange={handleOnPolicySelectChange}
+            data-test-subj={getTestId('policiesSelectable')}
+          >
+            {listBuilderCallback}
```

Comment:
I'm quite confused on how this works and what is it for?

Reply:
hahaha
The component seems to allow you to control how the children are layed out (thats my interpertation just based on the docs/coce). This call back receives 2 arguments - the `list` component and the `search` component if searchable is enabled.